### PR TITLE
[http-cache] follow up

### PR DESCRIPTION
**Pull request by Uzlopak**
follow up of http-cache pr

### CODE REVIEW

File: lib/handler/cache-handler.js [hunk 1]
```diff
@@ -75,14 +92,14 @@ class CacheHandler extends DecoratorHandler {
     )
 
     if (
-      UNSAFE_METHODS.includes(this.#req.method) &&
+      !this.#methods.includes(this.#requestOptions.method) &&
       statusCode >= 200 &&
       statusCode <= 399
     ) {
```
Comment (Uzlopak) [inline lib/handler/cache-handler.js:98]: This whole part seems strange. We do need to check

File: lib/handler/cache-handler.js [hunk 2]
```diff
@@ -75,14 +75,14 @@ class CacheHandler extends DecoratorHandler {
     )
 
     if (
-      UNSAFE_METHODS.includes(this.#req.method) &&
+      !this.#methods.includes(this.#requestOptions.method) &&
```
Comment (Uzlopak) [inline lib/handler/cache-handler.js:78]: WHy did we check for unsafe methods? this whole if condition seems actually kind of dubious.

Comment (Uzlopak) [reply lib/handler/cache-handler.js:78]: I would have expected 

```js
    if (
      !this.#methods.includes(this.#requestOptions.method) ||
      (
        statusCode < 200 &&
        statusCode >= 400
      )
    ) {
```

Comment (Uzlopak) [reply lib/handler/cache-handler.js:78]: @flakey5 

The invalidation logic is kind of strange. Also what is the point of passing methods, when they are ignored anyway? So I guess actually methods would have been an array of safe methods?

Comment (Uzlopak) [reply lib/handler/cache-handler.js:78]: > A cache MUST invalidate the target URI ([Section 7.1](https://www.rfc-editor.org/rfc/rfc9110#section-7.1) of [[HTTP](https://www.rfc-editor.org/rfc/rfc9111.html#HTTP)]) when it receives a non-error status code in response to an unsafe request method (including methods whose safety is unknown).

So I guess my proposed change should be right, as the passed methods will be considered safe methods. Especially in regard of the last sentence in the spec, that methods whose safety is unknown is also covered.

File: lib/interceptor/cache.js [hunk 3]
```diff
@@ -4,48 +4,40 @@ const util = require('../core/util')
...
-    }
-  } else {
-    globalOpts.methods = ['GET']
+  const globalOpts = {
+    store,
+    methods
   }
 
-  // Safe methods the user wants and unsafe methods
-  const methods = [...globalOpts.methods, ...UNSAFE_METHODS]
```
Comment (Uzlopak) [inline lib/interceptor/cache.js:39]: why did me "mixed" with unsafe methods?

this would result in caching unsafe requests

**Comment by Uzlopak** [comment]
@flakey5 PTAL

File: lib/interceptor/cache.js (hunk 3 above)
Comment (flakey5) [reply lib/interceptor/cache.js:39]: this allows the unsafe methods to get to the cache handler, where they'll cause a cache purge if the response is a success

**Review by metcoder95 — APPROVED** [review]

**Pull request merged**